feat: add exponential backoff for sending data (#76)
SumoLogic/fluentd-output-sumologic: README.md

* `compress_encoding` - Compression encoding format, either `gzip` or `deflate` (default `gzip`)
* `custom_fields` - Comma-separated key=value list of fields to apply to every log. [more information](https://help.sumologic.com/Manage/Fields#http-source-fields)
* `custom_dimensions` - Comma-separated key=value list of dimensions to apply to every metric. [more information](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers)

> * `retry_forever` - If yes, plugin will be sending data until it sends it successfully

Author: I think I will remove it, as `retry_forever` is equal to `retry_timeout`=`0` and `retry_max_times`=0

* `compress_encoding` - Compression encoding format, either `gzip` or `deflate` (default `gzip`)
* `custom_fields` - Comma-separated key=value list of fields to apply to every log. [more information](https://help.sumologic.com/Manage/Fields#http-source-fields)
* `custom_dimensions` - Comma-separated key=value list of dimensions to apply to every metric. [more information](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers)
* `skip_retry` - Disable custom retry mechanism. As this is `true` by default due to backward compatibility,
  we recommend to disable it and configure following parameters (`retry_min_interval`, `retry_max_interval`, `retry_timeout`, `retry_max_times`)
* `retry_min_interval` - Minimum interval to wait between sending tries (default is `1s`)
* `retry_max_interval` - Maximum interval to wait between sending tries (default is `5m`)

> * `retry_timeout` - Time after which the data is going to be dropped (default is `72h`) (`0s` means that this argument will be ignored)

Reviewer: ```suggestion
* `retry_timeout` - Time after which the data is going to be dropped (default is `72h`) (`0s` means that there is no timeout)
```

> * `retry_max_times` - Maximum number of retries (default is `0`) (`0` means that this argument will be ignored)

Reviewer: ```suggestion
* `retry_max_times` - Maximum number of retries (default is `0`) (`0` means that there is no max retry times, retries will happen forever)
```

}
```

## Retry Mechanism

`retry_min_interval`, `retry_max_interval`, `retry_timeout`, `retry_max_times` are not the [buffer retries parameters][buffer_retries].
Due to technical reason, this plugin implements it's own retrying back-off exponential mechanism.

[buffer_retries]: https://docs.fluentd.org/configuration/buffer-section#retries-parameters

> 

Author: @astencel-sumo is it sufficient?

Reviewer: Thanks!
